Add new bindip argument to CLI for serve

cody-code-wy opened the pull request:
This allows binding to a specific IP on a machine with multiple IP addresses, or binding only to localhost when running for local use such as with a reverse proxy or for testing.

Also update all tests for the new method signature. either passing an empty string when nothing will be served, or localhost when serve is used.

I am sorry that I have not added any new tests here, but I'm not entirely sure how to create any kind of useful test for this feature, and it is fortunately extremely simple in it's implementation. It does work as expected in my manual testing.

-----

I find this particularly useful as I often do development on a Mac which causes a security popup anytime a new executable attempts to bind to any non-loopback address, and since go in compiled that is every time I make any change.

Also I often work in environments with multiple IP addresses per machine and want to allow applications only to listen on specific IP addresses. This can be because multiple applications want to use the same port number, or just for security reasons like adding stunnel to wrap an API in HTTPS

calvinmclean commented:
Thanks for the contribution! This looks like a good improvement. Could you also fully combine/replace the `port` value with the `bindip` (or `bindaddress` now) since `http.Server` expects a string in the form `host:port` anyways? It is also compatible with #29 already

cody-code-wy commented:
I rebased on master to ensure that it is fully compatible with any changes, and everything works as expected.

I combined the port and bindIP to just bindAddress and updated the tests for the new signature. in manual testing everything works.

The CI test for the PR failed, but it seems to be unrelated to this PR see issue #30 for a possible explanation. On my machine the tests pass most of the time.
